### PR TITLE
Add generic UDP line-protocol service input

### PR DESCRIPTION
**Pull request by whatyouhide**
@lexmag and I added the UDP line-protocol service input discussed in #481. The PR fully implements the UDP service input but is still lacking tests and a README; we're working on both but we would like some feedback on the current implementation :).

### CODE REVIEW

File: plugins/inputs/udp_line_protocol/udp_line_protocol.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+func (u *UdpLineProtocol) SampleConfig() string {
+	return sampleConfig
+}
+
+func (u *UdpLineProtocol) Description() string {
+	return "Read metrics from one or more commands that can output to stdout"
+}
+
+func (u *UdpLineProtocol) Start(_ telegraf.Accumulator) error {
```
Comment (sparrc) [inline plugins/inputs/udp_line_protocol/udp_line_protocol.go]: lock/unlock the start function

**Comment by sparrc** [comment]
thanks @whatyouhide, a few things that will need to be done to complete this:

1. Your Start function should utilize the accumulator by continuously adding points that enter, rather than filling an internal buffer and flushing on every call to Gather. See the [mqtt_consumer plugin](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/mqtt_consumer/mqtt_consumer.go) as an example of this (Gather will become a dummy function)
1. After doing the above, there should no longer be a need for keeping an internal buffer and dropping any metrics.
1. Your listener should not be specific to "line protocol", it should utilize the generic data format parsers (also see mqtt_consumer or [exec](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/exec/exec.go)). There is a short guide in the README explaining how to utilize the generic parsers [here](https://github.com/influxdata/telegraf/blob/master/CONTRIBUTING.md#input-plugins-accepting-arbitrary-data-formats)

**Comment by whatyouhide** [comment]
@sparrc I updated the PR with your suggested changes, so this is ready for review (I'll start writing tests right now).

**Comment by whatyouhide** [comment]
@sparrc I added a README and tests as well. Let me know if I can do anything else :D

File: plugins/inputs/udp_listener/udp_listener.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	for {
+		select {
+		case <-u.done:
+			return nil
+		case packet := <-u.in:
+			metrics, err := u.parser.Parse(packet)
+			if err == nil {
+				u.storeMetrics(metrics)
+			} else {
+				log.Printf("Malformed packet")
```
Comment (sparrc) [inline plugins/inputs/udp_listener/udp_listener.go]: print the packet and the error message here:

```
log.Printf("Malformed packet, packet: [%s], Error: %s", packet, err)
```

Comment (whatyouhide) [inline plugins/inputs/udp_listener/udp_listener.go]: @sparrc updated and rebased!

**Comment by whatyouhide** [comment]
I suspect the failings on CircleCI are not related to the changes in the PR. Let me know if they are, I'll take a deeper look at them and see what I can fix.

**Comment by sparrc** [comment]
this was related to #745, which I've just closed, could you rebase your changes and re-push?

**Comment by whatyouhide** [comment]
@sparrc rebased over master. :)

**Comment by sparrc** [comment]
great! thanks @whatyouhide, LGTM :+1:

**Comment by whatyouhide** [comment]
@sparrc would it be a problem to have a release once this gets merged? We're about to start using telegraf in my company and a package instead of building from source would be nice :).

**Comment by sparrc** [comment]
yes, sure, shouldn't be a problem

**Comment by sparrc** [comment]
@whatyouhide I've merged this into master, with a couple changes:

1. Changed default port to 8092, this was to fit in more with other TICK stack things (InfluxDB uses 8090 by default for udp, Kapacitor uses 9092). I also didn't want the default to conflict with the statsd default.
1. Added a note on OS UDP buffer sizes in the README.

hope those are both OK, and thanks for the contribution!

**Comment by whatyouhide** [comment]
@sparrc of course they're OK :) thanks for merging!

**Comment by lexmag** [comment]
Hey @sparrc, any ETA on version with this input?

**Comment by whatyouhide** [comment]
Hey @sparrc, sorry for bothering you again! Any news on the telegraf release that would include this PR? Thanks :heart_decoration:

**Comment by sparrc** [comment]
next week we will do 0.11 and it will have this plugin

**Comment by whatyouhide** [comment]
Awesome, thanks so much! :heart_eyes:

**Pull request merged**